Update solidity-by-example.rst

varunagarwal315 opened the pull request:
Might be trivial, but makes more sense to be able to directly return the name of the winner for the election. If the position of the winner on the proposal[] array is returned, then people still don't know the name of the person who won.

varunagarwal315 committed:
Update solidity-by-example.rst

Might be trivial, but makes more sense to be able to directly return the name of the winner for the election. If the position of the winner on the proposal[] array is returned, then people still don't know the name of the person who won.

pirapira reviewed (changes requested):
I think the new function can call `winningProposal()`, and then the code is much simpler.  Currently in this PR `winningProposal()` and `winnerName()` contain largely duplicate code.

varunagarwal315 committed:
Update solidity-by-example.rst

made changes suggested to simplify code. Hope this is enough

Thanks

varunagarwal315 commented:
Removed all the duplicate code as suggested and simply called the winningProposal() function. Added some comments to help explain the function to new users. Thanks pirapira for the suggestions

pirapira commented on docs/solidity-by-example.rst:
```diff
@@ -170,6 +170,15 @@ of votes.
+        
+        //Calls winningProposal() function to get the index
+        //of the winner contained in the proposals array and then
+        //returns the name of the winner
```
Just for uniformity, will you add a space between `//` and the following text?

varunagarwal315 committed:
Update solidity-by-example.rst

added the space. Sorry, just slipped my mind.

pirapira commented:
Thank you!  Let me compile it...

pirapira commented:
@chriseth is there a particular reason why we do not allow returning structs?  Or shall I create an issue about that?

VoR0220 commented:
@pirapira there is an issue for this, but basically the capability currently does not exist due to the ABI needing to be defined for structs.

varunagarwal315 commented:
@pirapira Are there any further changes to be made by me before this can be merged? Or is the change too trivial for a merge with dev branch.

Thanks

pirapira commented:
@varunagarwal315 no, no, this was just a pause to hear people.
